### ISSUE TITLE
Error: Starting failed in GOG integration settings.

### ISSUE THREAD

**Issue by LeonardFiedrowicz** [UncleGoogle/galaxy-integration-humblebundle#27]
**Describe the bug**
Plugin fails to launch according to the settings screen in GOG. Games appear in the list in GOG but the install/launch button can't be clicked because.

**Screenshots**
![GalaxyClient_KPvJeKhLCx](https://user-images.githubusercontent.com/50417119/64813500-cb5f9280-d5a1-11e9-8119-c94d690c4193.png)

**Attach logs**
2019-09-12 21:12:21,902 - root - ERROR - Task exception was never retrieved
future: <Task finished coro=<StreamWriter.drain() done, defined at D:\obj\Windows-Release\37win32_Release\msi_python\zip_win32\streams.py:328> exception=ConnectionResetError('Connection lost')>
Traceback (most recent call last):
  File "D:\obj\Windows-Release\37win32_Release\msi_python\zip_win32\streams.py", line 348, in drain
  File "D:\obj\Windows-Release\37win32_Release\msi_python\zip_win32\streams.py", line 202, in _drain_helper
ConnectionResetError: Connection lost
2019-09-12 21:12:21,922 - root - ERROR - Task exception was never retrieved
future: <Task finished coro=<StreamWriter.drain() done, defined at D:\obj\Windows-Release\37win32_Release\msi_python\zip_win32\streams.py:328> exception=ConnectionResetError('Connection lost')>
Traceback (most recent call last):
  File "D:\obj\Windows-Release\37win32_Release\msi_python\zip_win32\streams.py", line 348, in drain
  File "D:\obj\Windows-Release\37win32_Release\msi_python\zip_win32\streams.py", line 202, in _drain_helper
ConnectionResetError: Connection lost

[plugin-humble-f0ca3d80-a432-4d35-a9e3-60f27161ac3a.log](https://github.com/UncleGoogle/galaxy-integration-humblebundle/files/3607116/plugin-humble-f0ca3d80-a432-4d35-a9e3-60f27161ac3a.log)

**Comment by UncleGoogle** [comment]
Hi, thanks for the report. Do you mean latest Galaxy update or latest plugin update? I do not recognize what is happening - will take a closer look when return from vacation.

**Comment by Eremeir** [comment]
Newest plugin commit. It's been like this before, and after the Galaxy update.

**Comment by LeonardFiedrowicz** [comment]
It actually doesnt seem to work with any version for me now...it worked before.

**Comment by Nipponshin** [comment]
Can confirm that this is also happening to me.

**Comment by UncleGoogle** [comment]
There are 2 issues:
- serious bug in integration: cache was replicated instead of replacing the old one. Therefore, every few minutes/ every Galaxy launch it doubled cache size... and finally it was too big because...:
- edge-case/optimization problem in api: a blocking loop when reading very long, one-line text from cache. So when cache was getting big (few mb) - Galaxy treats integration as "dead"/blocking and kills it.

Guys, please test fixed version. 
If anyone has a really big humble library - please send fixed version logs to see if it is worth to fix the second issue in API.

[humblebundle_0.4.1rc.zip](https://github.com/UncleGoogle/galaxy-integration-humblebundle/files/3628482/humblebundle_0.4.1rc.zip)

**Comment by Nipponshin** [comment]
Can confirm that the RC works but I had to manually delete the DB, retrying from the launcher did not work. Do you still need the logs? I am not sure from where to grab them, can you point me in the right direction?

**Comment by Eremeir** [comment]
![unknown](https://user-images.githubusercontent.com/20992519/65189639-2af0ee80-da37-11e9-8be3-f5a926b9ae2a.png)
This just seemed funny to me.
Deleting the db was required to fix the version on my end too. Fixed db size is 1.82MB.
No issues with the plugin on my end. Is that big enough to potentially experience the second bug?

**Comment by Xiuhtec** [comment]
Release candidate works well here. Cached DB size was ~8MB before and 3.6MB now. For reference I have 692 games imported (and 73 of those show up as "Unknown game", trying to find a less annoying way than manual list comparison to determine which aren't being identified correctly to report it to CDPR as I know it's not your bug).

**Comment by UncleGoogle** [comment]
Ok, I've mocked big cache and tested it once again. Automigration doesn't work as Galaxy ignores any request after initial, too-long cache read. **So the only solution is to remove old database (or just remove "library" key from this .db file).**

For my laptop, the upper cache size limit is somewhere between 3 and 5MB. Not much. Tried to fix api, but looks like the code is ok (I mean still slow, but not blocking). Galaxy just don't want to wait over 10seconds for this step. It may be a problem for slower machines/bigger cache.
